src/promptflow/promptflow/_utils/tool_utils.py

```diff
@@ -305,6 +306,33 @@ def load_function_from_function_path(func_path: str):
         )
 
 
+def assign_tool_input_index_for_ux_order_if_needed(tool):
+    tool_type = tool.get("type")
+    if should_preserve_tool_inputs_order(tool_type) and "inputs" in tool:
+        inputs_dict = tool["inputs"]
+        input_index = 0
+        for input_name, settings in inputs_dict.items():
```

Comment:
Add comments why this dict iterator can keep the order? I remember python builtin dict object has no order.

Reply:
Added, thanks.